manager/api/jpa/src/main/java/io/apiman/manager/api/jpa/JpaStorage.java

```diff
@@ -2514,17 +2505,17 @@ private void deleteAllPolicies(ClientBean clientBean) throws StorageException {
     }
 
     private void deleteAllPolicies(OrganizationBean organizationBean, String entityId, PolicyType type) throws StorageException {
-        String jpql =
-            "DELETE FROM PolicyBean b "
-          + "      WHERE b.organizationId = :orgId ";
+        DeleteCriteriaBuilder<PolicyBean> builder = getCriteriaBuilderFactory()
+                .delete(getActiveEntityManager(), PolicyBean.class, "b")
+                .where("b.organizationId").eq(organizationBean.getId());
 
-        if (entityId != null && type != null) {
-            jpql += String.format("AND b.entityId = '%s' AND b.type = '%s' ", entityId, type.name());
+        if (entityId != null) {
```

Comment:
Im pretty sure that the delete methods which contain optional entityId and entityType are always called with missing entityId and type or with both of them set.
We could merge the two if's, i think

Reply:
Thanks for the review!

I was thinking it could be useful to have a query that allows each of those individually (even if we aren't using it quite yet). I should probably label them nullable and/or make overloaded methods with fewer parameters.